### PR TITLE
Fix /latest query not honoring cache if channel is empty, causing an …

### PR DESCRIPTION
**Pull request by dkleckler**
…S3 hit.

### CODE REVIEW

File: src/test/java/com/flightstats/hub/dao/aws/ClusterContentServiceTest.java [hunk 1]
```diff
@@ -254,6 +283,10 @@ private ContentKey stubCacheLatest(int ageInSeconds, String keyName) {
         return cachedKey;
     }
 
+    private void stubCachedEmptyChannelLatest() {
```
Comment (manjula-chikkanagappa) [inline src/test/java/com/flightstats/hub/dao/aws/ClusterContentServiceTest.java]: Method is overkill to mock stuff. I would add comments instead

File: src/test/java/com/flightstats/hub/dao/aws/ClusterContentServiceTest.java [hunk 2]
```diff
@@ -17,6 +17,7 @@
 import com.flightstats.hub.util.HubUtils;
 import com.flightstats.hub.util.TimeUtil;
 import lombok.SneakyThrows;
+import org.joda.time.DateTime;
```
Comment (manjula-chikkanagappa) [inline src/test/java/com/flightstats/hub/dao/aws/ClusterContentServiceTest.java]: Java 8 date library should be favored here. Performance is better with Java 8 dates compared to joda-time.

Comment (lkemmerer) [reply src/test/java/com/flightstats/hub/dao/aws/ClusterContentServiceTest.java]: I believe we have a task to migrate the hub off joda and onto Java 8 native stuff, but it'll be a pretty big and sensitive change. I'll check to make sure we're tracking that debt.

Comment (dkleckler) [reply src/test/java/com/flightstats/hub/dao/aws/ClusterContentServiceTest.java]: Yeah, joda's heavily used throughout hub, so leaving this for now for consistency.  Agree it'd be nice to switch over though.

**Pull request merged**